src/storage/src/sink/kafka.rs

```diff
@@ -334,43 +402,58 @@ struct ProgressInitState {
 }
 
 impl ProgressInitState {
-    fn to_running(self, gate_ts: Rc<Cell<Option<Timestamp>>>) -> ProgressRunningState {
+    fn to_running(
+        self,
+        gate_ts: Rc<Cell<Option<Timestamp>>>,
+        healthchecker: Arc<Mutex<Option<Healthchecker>>>,
+    ) -> ProgressRunningState {
         ProgressRunningState {
             topic: self.topic,
             key: self.key,
             gate_ts,
+            healthchecker,
         }
     }
 }
 
-#[derive(Debug, Clone)]
+#[derive(Debug)]
 struct ProgressRunningState {
     topic: String,
     key: String,
     gate_ts: Rc<Cell<Option<Timestamp>>>,
+    // Using Arc<Mutex> because clippy complains about holding a RefCell reference across await points if we use
+    // an `Rc<RefCell>`.
+    healthchecker: Arc<Mutex<Option<Healthchecker>>>,
```

Comment:
Does this need to live in the progress state? It would be nice if we could restrict this to only managing progress-topic-relevant stuff.

Feels like we might be able to share it the same way we share state like `shutdown_flush` or `metrics` in `produce_to_kafka`?

Reply:
hmm -- I put it here, because, like the `gate_ts`, it's not something we're able to construct until just before we enter the running state -- and they're things that we definitely want to construct (well, try to anyway, given the Option) before moving into the running state

That being said, you do make a good point.  What do you think of pushing to until we tackle the 
> It's becoming increasingly clear that we'll want to reorganize the KafkaSinkState struct at some point

I mentioned in the description?

Reply:
> it's not something we're able to construct until just before we enter the running state.

why's this? feels like we'd want to initialize it early in the lifecycle, so we can promptly report the `starting` state.